src/Microsoft.ApplicationInsights/Metric.cs

```diff
@@ -1200,17 +1200,20 @@ public bool TryTrackValue(object metricValue, string dimension1Value, string dim
         }
        
         private static void EnsureConfigurationValid(
-                                    bool isMultidimensional,
+                                    int dimensionsCount,
                                     MetricConfiguration configuration)
         {
             Util.ValidateNotNull(configuration, nameof(configuration));
             Util.ValidateNotNull(configuration.SeriesConfig, nameof(configuration.SeriesConfig));
 
-            if (isMultidimensional && configuration.ValuesPerDimensionLimit < 1)
+            for (int d = 0; d < dimensionsCount; d++)
             {
-                throw new ArgumentException("Multidimensional metrics must allow at least one dimension-value per dimesion"
-                                          + Invariant($" (but {configuration.ValuesPerDimensionLimit} was specified")
-                                          + Invariant($" in {nameof(configuration)}.{nameof(configuration.ValuesPerDimensionLimit)})."));
+                if (configuration.GetValuesPerDimensionLimit(d + 1) < 1)
+                {
+                    throw new ArgumentException("Multidimensional metrics must allow at least one dimension-value per dimesion"
```

Comment:
again - it will be great if you can find a way to not throw from the public API call. Maybe error message and fall back logic.

Reply:
I remember the prev conversation, just did not get to it yet. Note - the diff shows this as addition, but the code was there before.
It will be easier at some places and harder at others. It is not always possible to replace nonsensical parameters with reasonable defaults.